packages/vite/src/node/server/hmr.ts

```diff
@@ -228,7 +228,8 @@ export function updateModules(
   const updates: Update[] = []
   const invalidatedModules = new Set<ModuleNode>()
   const traversedModules = new Set<ModuleNode>()
-  let needFullReload: HasDeadEnd = false
+  // This should be possible only when a root module is invalidated via import.meta.hot.invalidate()
```

Comment:
```suggestion
  // Modules could be empty if a root module is invalidated via import.meta.hot.invalidate()
```

Updated this a bit so the comment doesn't refer to `needsFullReload`.